CONTRIBUTING.md

```diff
@@ -1,24 +1,181 @@
-### Contributing to Qt
+## Contributing to Qt
 
-To contribute, fork this project and submit a pull-request.
+Thanks for taking the time to contribute!
 
-Your code will be reviewed and merged once it:
+In here you'll find a series of guidelines for how you can make Qt.py better suit your needs and the needs of the target audience - film, games and tv.
 
-1. Does something useful
-1. Is up to par with surrounding code
+Qt.py was born to address the growing needs in these industries for the development of software capable of running with more than a single flavour of the Qt bindings for Python - PySide, PySide2, PyQt4 and PyQt5.
+
+**Table of contents**
+
+- [Development goals](#development-goals)
+  - [Support co-existence](#support-co-existence)
+  - [Don't get smart](#dont-get-smart)
+  - [No bugs](#no-bugs)
+- [How can I contribute?](#how-can-i-contribute)
+  - [Reporting Bugs](#reporting-bugs)
+  - [Suggesting Enhancements](#suggesting-enhancements)
+  - [Your First Code Contribution](#your-first-code-contribution)
+  - [Pull Requests](#pull-requests)
+- [Style](#style)
+  - [Commits](#commits)
+  - [Version bumping](#version-bumping)
+  - [Making a release](#making-a-release)
+
+<br>
+
+### Development Goals
+
+Tha goal for us developers, in a nutshell, is this.
+
+1. **Support co-existence** - Qt.py should not affect other bindings running in same interpreter session.
+1. **Don't get smart** - One file, copy/paste installation, keep it simple.
+1. **No bugs** - No implementations == No bugs.
+
+Each of these deserve some explanation and rationale.
+
+<br>
+
+##### Support co-existance
+
+Importing or otherwise using Qt.py *cannot* break other bindings. The reason being that our userbase frequently runs multiple applications, some of them using the original binding, in the same interpreter session.
+
+```python
+# Wrong
+old_translate_fn = QtWidgets.QApplication.translate
+
+def translate(context, key, disambiguation=None, encoding=None, n=0):
+    return old_translate_fn(context, key, disambiguation, n)
+
+# Overwrite original with an incompatible version
+QtWidgets.QApplication.translate = staticmethod(translate)
+```
+
+```python
+# Right
+...
+QtWidgets.QApplication.translate_ = staticmethod(translate)
+```
 
-Development for this project takes place in individual forks. The parent project ever only contains a single branch, a branch containing the latest working version of the project.
+<br>
+
+##### Don't get smart
+
+At the end of the day, Qt.py is a middle-man. It delegates request you make to the appropriate reciever, such as PySide2. There are many ways in which this delegation can go, and Python - being a capable dynamic programming language - can easily make something like this simple high-level goal into a tangled mess of automatic and optimised nest of bugs.
```

Comment:
> It delegates request you make to the appropriate reciever, such as PySide2.

It delegates **the** request you make to the appropriate reciever, such as PySide2.